[2024-03-19 21:02:36,814: INFO: main: Welcome to the custom log]
[2024-03-20 17:18:48,249: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-20 17:18:48,259: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-20 17:20:15,946: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-20 17:20:15,947: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-20 17:37:57,799: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-20 17:37:57,810: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-20 17:37:57,812: INFO: common: created directory at: artifacts]
[2024-03-20 17:37:57,813: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-20 17:37:57,814: INFO: 723630791: Downloading data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-20 17:40:57,891: INFO: 723630791: Downloaded data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-20 21:18:48,235: INFO: main: This is the main output]
[2024-03-20 22:48:12,016: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-03-20 22:48:12,020: ERROR: main: name 'yaml' is not defined]
Traceback (most recent call last):
  File "main.py", line 10, in <module>
    obj.main()
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/pipeline/stage_01_data_ingestion.py", line 13, in main
    config = ConfigurationManager()
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/config/configuration.py", line 16, in __init__
    self.config = read_yaml(config_filepath)
  File "/home/oadebayo/.local/lib/python3.8/site-packages/ensure/main.py", line 849, in __call__
    return_val = self.f(*args, **kwargs)
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/utils/common.py", line 37, in read_yaml
    raise e
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/utils/common.py", line 31, in read_yaml
    content = yaml.safe_load(yaml_file)
NameError: name 'yaml' is not defined
[2024-03-21 00:02:40,469: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-03-21 00:02:40,478: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 00:02:40,479: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 00:02:40,479: INFO: common: created directory at: artifacts]
[2024-03-21 00:02:40,479: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-21 00:02:40,479: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 00:06:35,070: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 00:06:37,147: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-03-21 02:20:17,508: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-03-21 02:20:17,525: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 02:20:17,528: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 02:20:17,530: INFO: common: created directory at: artifacts]
[2024-03-21 02:20:17,531: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-21 02:20:17,531: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 02:22:17,349: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 02:22:28,110: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-03-21 15:57:29,443: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 15:57:29,446: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 15:57:29,453: INFO: common: created directory at: artifacts]
[2024-03-21 15:57:29,454: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 15:57:30,462: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 16:01:54,491: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 16:01:54,495: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 16:01:54,496: INFO: common: created directory at: artifacts]
[2024-03-21 16:01:54,496: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 16:02:08,938: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 16:02:08,941: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 16:02:08,942: INFO: common: created directory at: artifacts]
[2024-03-21 16:02:08,943: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 16:02:09,543: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 16:05:12,720: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 16:05:12,723: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 16:05:12,724: INFO: common: created directory at: artifacts]
[2024-03-21 16:05:12,725: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 16:05:13,589: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 16:08:35,065: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 16:08:35,069: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 16:08:35,070: INFO: common: created directory at: artifacts]
[2024-03-21 16:08:35,071: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 16:08:35,643: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 16:10:39,441: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 16:10:39,445: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 16:10:39,446: INFO: common: created directory at: artifacts]
[2024-03-21 16:10:39,446: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 16:10:40,139: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 18:58:21,136: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-03-21 18:58:21,140: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 18:58:21,144: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 18:58:21,144: INFO: common: created directory at: artifacts]
[2024-03-21 18:58:21,145: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-21 18:58:21,145: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 19:00:17,975: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/1enFGVoZhyqs81zUBAUfEmhhWqez1KFEA/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 19:00:28,567: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-03-21 19:00:28,567: INFO: main: *******************]
[2024-03-21 19:00:28,567: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-03-21 19:00:28,570: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 19:00:28,585: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 19:00:28,588: INFO: common: created directory at: artifacts]
[2024-03-21 19:00:28,589: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 19:00:29,884: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 19:00:31,693: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-03-21 20:19:42,875: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 20:19:42,879: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 20:19:42,880: INFO: common: created directory at: artifacts]
[2024-03-21 20:21:26,123: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 20:21:26,137: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 20:21:26,140: INFO: common: created directory at: artifacts]
[2024-03-21 20:25:00,067: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 20:25:00,087: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 20:25:00,105: INFO: common: created directory at: artifacts]
[2024-03-21 20:25:00,106: INFO: common: created directory at: artifacts/model_training]
[2024-03-21 22:47:14,164: INFO: main: >>>>>> stage Data Ingestion stage started <<<<<<]
[2024-03-21 22:47:14,172: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 22:47:14,177: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 22:47:14,180: INFO: common: created directory at: artifacts]
[2024-03-21 22:47:14,180: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-21 22:47:14,181: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1ALX99Nuez1VHrQFm3UZRQrSZrrkEKrky/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 22:49:32,148: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/1ALX99Nuez1VHrQFm3UZRQrSZrrkEKrky/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 22:49:37,527: INFO: main: >>>>>> stage Data Ingestion stage completed <<<<<<

x==========x]
[2024-03-21 22:49:37,527: INFO: main: *******************]
[2024-03-21 22:49:37,528: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-03-21 22:49:37,530: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 22:49:37,532: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 22:49:37,533: INFO: common: created directory at: artifacts]
[2024-03-21 22:49:37,533: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 22:49:38,469: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 22:49:40,195: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-03-21 22:53:03,123: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 22:53:03,127: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 22:53:03,129: INFO: common: created directory at: artifacts]
[2024-03-21 22:53:03,133: INFO: common: created directory at: artifacts/model_training]
[2024-03-21 22:54:49,653: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 22:54:49,657: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 22:54:49,658: INFO: common: created directory at: artifacts]
[2024-03-21 22:54:49,659: INFO: common: created directory at: artifacts/model_training]
[2024-03-21 23:24:18,297: INFO: main: >>>>>> stage Data ingestion started <<<<<<]
[2024-03-21 23:24:18,298: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 23:24:18,299: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 23:24:18,299: INFO: common: created directory at: artifacts]
[2024-03-21 23:24:18,299: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-21 23:24:18,299: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1ALX99Nuez1VHrQFm3UZRQrSZrrkEKrky/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 23:27:16,986: INFO: data_ingestion: Downloaded data from https://drive.google.com/file/d/1ALX99Nuez1VHrQFm3UZRQrSZrrkEKrky/view?usp=sharing into file artifacts/data_ingestion/data.zip]
[2024-03-21 23:27:18,950: INFO: main: >>>>>> stage Data ingestion completed <<<<<<

x==========x]
[2024-03-21 23:27:18,950: INFO: main: *******************]
[2024-03-21 23:27:18,950: INFO: main: >>>>>> stage Prepare base model started <<<<<<]
[2024-03-21 23:27:18,951: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 23:27:18,952: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 23:27:18,952: INFO: common: created directory at: artifacts]
[2024-03-21 23:27:18,952: INFO: common: created directory at: artifacts/prepare_base_model]
[2024-03-21 23:27:19,243: WARNING: saving_utils: Compiled the loaded model, but the compiled metrics have yet to be built. `model.compile_metrics` will be empty until you train or evaluate the model.]
[2024-03-21 23:27:19,645: INFO: main: >>>>>> stage Prepare base model completed <<<<<<

x==========x]
[2024-03-21 23:27:19,645: INFO: main: *******************]
[2024-03-21 23:27:19,645: INFO: main: >>>>>> stage Model training started <<<<<<]
[2024-03-21 23:27:19,647: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-21 23:27:19,647: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-21 23:27:19,648: INFO: common: created directory at: artifacts]
[2024-03-21 23:27:19,648: ERROR: main: "'ConfigBox' object has no attribute 'training'"]
Traceback (most recent call last):
  File "box/box.py", line 503, in box.box.Box.__getitem__
KeyError: 'training'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 536, in box.box.Box.__getattr__
  File "box/box.py", line 524, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'training'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 538, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'training'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/config_box.py", line 28, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 552, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'training'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 503, in box.box.Box.__getitem__
KeyError: 'training'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "box/box.py", line 536, in box.box.Box.__getattr__
  File "box/box.py", line 524, in box.box.Box.__getitem__
box.exceptions.BoxKeyError: "'training'"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "box/box.py", line 538, in box.box.Box.__getattr__
AttributeError: 'ConfigBox' object has no attribute 'training'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "main.py", line 35, in <module>
    model_training_obj.main()
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/pipeline/stage_03_model_training.py", line 14, in main
    training_config = config.get_training_config()
  File "/home/oadebayo/PhD/DL_Benign_vs_Malignant_skin_tumour_with_keloid/src/cnnClassifier_benign_vs_malignant/config/configuration.py", line 62, in get_training_config
    training = self.config.training
  File "box/config_box.py", line 30, in box.config_box.ConfigBox.__getattr__
  File "box/box.py", line 552, in box.box.Box.__getattr__
box.exceptions.BoxKeyError: "'ConfigBox' object has no attribute 'training'"
[2024-03-22 00:22:08,830: INFO: main: >>>>>> stage Data ingestion started <<<<<<]
[2024-03-22 00:22:08,831: INFO: common: yaml file: config/config.yaml loaded successfully]
[2024-03-22 00:22:08,832: INFO: common: yaml file: params.yaml loaded successfully]
[2024-03-22 00:22:08,832: INFO: common: created directory at: artifacts]
[2024-03-22 00:22:08,832: INFO: common: created directory at: artifacts/data_ingestion]
[2024-03-22 00:22:08,832: INFO: data_ingestion: Downloading data from https://drive.google.com/file/d/1ALX99Nuez1VHrQFm3UZRQrSZrrkEKrky/view?usp=sharing into file artifacts/data_ingestion/data.zip]
